webviz_subsurface/plugins/_simulation_time_series/views/subplot_view/_subplot_view.py

```diff
@@ -0,0 +1,900 @@
+
+from webviz_subsurface._providers import Frequency
+from webviz_subsurface._utils.unique_theming import unique_colors
+from webviz_subsurface._utils.vector_calculator import get_selected_expressions
+
+from ._property_serialization import (
+    EnsembleSubplotBuilder,
+    GraphFigureBuilderBase,
+    VectorSubplotBuilder,
+)
+from ._settings._ensembles import EnsemblesSettings
+from ._settings._filter_realization import FilterRealizationSettings
+from ._settings._group_by import GroupBySettings
+from ._settings._resampling_frequency import ResamplingFrequencySettings
+from ._settings._time_series import TimeSeriesSettings
+from ._settings._visualization import VisualizationSettings
+from ._types import (
+    DeltaEnsemble,
+    FanchartOptions,
+    StatisticsFromOptions,
+    StatisticsOptions,
+    SubplotGroupByOptions,
+    TraceOptions,
+    VisualizationOptions,
+)
+from ._utils import DerivedVectorsAccessor, ProviderSet, datetime_utils
+from ._utils.derived_ensemble_vectors_accessor_utils import (
+    create_derived_vectors_accessor_dict,
+)
+from ._utils.from_timeseries_cumulatives import (
+    datetime_to_intervalstr,
+    is_per_interval_or_per_day_vector,
+)
+from ._utils.history_vectors import create_history_vectors_df
+from ._utils.provider_set_utils import create_vector_plot_titles_from_provider_set
+from ._utils.trace_line_shape import get_simulation_line_shape
+from ._utils.vector_statistics import create_vectors_statistics_df
+from ._view_elements.subplot import SubplotGraph
+
+
+class SubplotView(ViewABC):
+    # pylint: disable=too-few-public-methods
+    class Ids:
+        SUBPLOT = "subplot"
+        MAIN_COLUMN = "main-column"
+
+        ENSEMBLE_SETTINGS = "ensemble-settings"
+        FILTER_REALIZATION_SETTINGS = "filter-realization-settings"
+        GROUP_BY_SETTINGS = "group-by-settings"
+        RESAMPLING_FREQUENCY_SETTINGS = "resampling-frequency-settings"
+        TIME_SERIES_SETTINGS = "time-series-settings"
+        VISUALIZATION_SETTINGS = "visualization-settings"
+
+    # pylint: disable=too-many-arguments, too-many-locals
+    def __init__(
+        self,
+        custom_vector_definitions: dict,
+        custom_vector_definitions_base: dict,
+        disable_resampling_dropdown: bool,
+        initial_selected_vectors: List[str],
+        initial_vector_selector_data: list,
+        initial_visualization: VisualizationOptions,
+        input_provider_set: ProviderSet,
+        predefined_expressions: List[ExpressionInfo],
+        selected_resampling_frequency: Frequency,
+        vector_calculator_data: List,
+        vector_selector_base_data: list,
+        theme: WebvizConfigTheme,
+        user_defined_vector_definitions: Dict[str, VectorDefinition],
+        observations: dict,  # TODO: Improve typehint?
+        line_shape_fallback: str = "linear",
+    ) -> None:
+        super().__init__("Subplot View")
+
+        column = self.add_column(SubplotView.Ids.MAIN_COLUMN)
```

Comment:
Is this column accessed anywhere in a callback? If not, consider removing the id since it is not required.